### PR TITLE
feat: adds local_nvme_ssd_block_config to beta-public-cluster

### PR DESCRIPTION
**Pull request by florianMalbranque**

### CODE REVIEW

**Comment by florianMalbranque** [comment]
@gtsorbo , @ericyz , @Jberlinsky   would you please take a look ?

**Comment by florianMalbranque** [comment]
had to rebase again, 
@apeabody would you be able to trigger ci run again please ?

@gtsorbo , @ericyz , @Jberlinsky <ould you be able to approve please ?

File: autogen/main/README.md [hunk 1]
```diff
@@ -213,6 +213,7 @@ The node_pools variable takes the following parameters:
 | local_ssd_count | The amount of local SSD disks that will be attached to each cluster node and may be used as a `hostpath` volume or a `local` PersistentVolume.  | 0 | Optional |
 {% if beta_cluster %}
 | local_ssd_ephemeral_count | The amount of local SSD disks that will be attached to each cluster node and assigned as scratch space as an `emptyDir` volume. If unspecified, ephemeral storage is backed by the cluster node boot disk. | 0 | Optional |
+| local_nvme_ssd_count | Number of raw-block local NVMe SSD disks to be attached to the node.Each local SSD is 375 GB in size. If zero, it means no raw-block local NVMe SSD disks to be attached to the node. | 0 | Optional |
```
Comment (apeabody) [inline autogen/main/README.md:216]: nit: typo
```suggestion
| local_nvme_ssd_count | Number of raw-block local NVMe SSD disks to be attached to the node. Each local SSD is 375 GB in size. If zero, it means no raw-block local NVMe SSD disks to be attached to the node. | 0 | Optional |
```

File: autogen/main/README.md [hunk 2]
```diff
@@ -213,6 +213,7 @@ The node_pools variable takes the following parameters:
 | local_ssd_count | The amount of local SSD disks that will be attached to each cluster node and may be used as a `hostpath` volume or a `local` PersistentVolume.  | 0 | Optional |
 {% if beta_cluster %}
```
Comment (apeabody) [inline autogen/main/README.md:214]: Does this need to be restricted to `beta` provider?  It appears to be available in the GA provider.

Comment (florianMalbranque) [reply autogen/main/README.md]: Sure, it's GA now !

**Comment by florianMalbranque** [comment]
Thanks for your review @apeabody, it should be good now

**Comment by apeabody** [comment]
Hi @florianMalbranque - Looks like we need a quick `make build`, thanks!

File: autogen/main/README.md [hunk 3]
```diff
@@ -211,9 +211,8 @@ The node_pools variable takes the following parameters:
 | key | The key required for the taint | | Required |
 | logging_variant | The type of logging agent that is deployed by default for newly created node pools in the cluster. Valid values include DEFAULT and MAX_THROUGHPUT.  | DEFAULT | Optional |
 | local_ssd_count | The amount of local SSD disks that will be attached to each cluster node and may be used as a `hostpath` volume or a `local` PersistentVolume.  | 0 | Optional |
-{% if beta_cluster %}
```
Comment (apeabody) [inline autogen/main/README.md:214]: Looks like `local_ssd_ephemeral_count` is still beta in the provider, so let's keep the `if beta_cluster / endif` around it.

File: autogen/main/cluster.tf.tmpl [hunk 4]
```diff
@@ -847,6 +847,13 @@ resource "google_container_node_pool" "windows_pools" {
         local_ssd_count = ephemeral_storage_config.value
       }
     }
+
```
Comment (apeabody) [inline autogen/main/cluster.tf.tmpl:850]: We should pull the `dynamic "local_nvme_ssd_block_config"` out of the `if beta_cluster` block

```suggestion
{% endif %}

```

File: autogen/main/cluster.tf.tmpl [hunk 5]
```diff
@@ -847,6 +847,13 @@ resource "google_container_node_pool" "windows_pools" {
...
+    dynamic "local_nvme_ssd_block_config" {
+      for_each = lookup(each.value, "local_nvme_ssd_count", 0) > 0 ? [1] : []
+      content {
+        local_ssd_count = local_nvme_ssd_block_config.value
+      }
+    }
     {% endif %}
```
Comment (apeabody) [inline autogen/main/cluster.tf.tmpl:857]:
```suggestion
```

**Review by apeabody — CHANGES_REQUESTED** [review]
Thanks @florianMalbranque - A few final notes on the `if beta_cluster` blocks as `ephemeral_storage_config` is still beta.

**Comment by florianMalbranque** [comment]
Thanks again for your reviews @apeabody ! it should be good now, PTAL

**Pull request merged**